### PR TITLE
Feature lwdev 4063 client search service

### PR DESCRIPTION
**Pull request by igromyko**

### CODE REVIEW

File: src/Lykke.Service.ClientSearch/Program.cs [hunk 1]
```diff
@@ -1,53 +1,53 @@
...
+using System.IO;
+using System.Runtime.Loader;
+using System.Threading;
+using Microsoft.AspNetCore.Hosting;
+using Lykke.JobTriggers.Triggers;
+using System.Threading.Tasks;
+
+namespace Lykke.Service.ClientSearch
+{
+    public static class Program
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.ClientSearch/Program.cs]: Use code from template
https://github.com/LykkeCity/lykke.dotnettemplates/blob/master/Lykke.Job.LykkeJob/src/Lykke.Job.LykkeJob/Program.cs

File: src/Lykke.Service.ClientSearch/Startup.cs [hunk 2]
```diff
@@ -116,8 +120,18 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IApplica
                 // NOTE: Service not yet recieve and process requests here
 
                 await ApplicationContainer.Resolve<IStartupManager>().StartAsync();
-
                 await Log.WriteMonitorAsync("", $"Env: {Program.EnvInfo}", "Started");
+
+                Indexer indexer = ApplicationContainer.Resolve<Indexer>();
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.ClientSearch/Startup.cs]: Why this code not in StartupManager?

File: src/Lykke.Service.ClientSearch/Startup.cs [hunk 3]
```diff
@@ -116,8 +120,18 @@ public void Configure(IApplicationBuilder app, IHostingEnvironment env, IApplica
...
+                await Task.Factory.StartNew(
+                    async () => {
+                        await indexer.Initialize();
+
+                        _triggerHost = new TriggerHost(new AutofacServiceProvider(ApplicationContainer));
+                        _triggerHostTask = _triggerHost.Start();
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.ClientSearch/Startup.cs]: _triggerHostTask & _triggerHost  are used only here. Why they are fields?

File: src/Lykke.Service.ClientSearch/Modules/ServiceModule.cs [hunk 4]
```diff
@@ -40,35 +40,34 @@ protected override void Load(ContainerBuilder builder)
...
-
-
-
-            builder.RegisterInstance<ITriggerManager>(new TriggerManager(new AutofacServiceProvider(_applicationContainer)));
-            builder.RegisterInstance<IPersonalDataService>(new PersonalDataService(_pdClientSettings.CurrentValue, _log));
-
+                .As<IShutdownManager>();
+
+
+            builder.RegisterInstance<IPersonalDataService>(new PersonalDataService(_pdClientSettings.CurrentValue, _log));
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.ClientSearch/Modules/ServiceModule.cs]: PersonalDataService can be registered as builder.RegisterType< PersonalDataService >().As< IPersonalDataService >()

File: src/Lykke.Service.ClientSearch/Program.cs (hunk 1 above)
Comment (igromyko) [reply src/Lykke.Service.ClientSearch/Program.cs]: replaced from template

File: src/Lykke.Service.ClientSearch/Startup.cs (hunk 2 above)
Comment (igromyko) [reply src/Lykke.Service.ClientSearch/Startup.cs]: removed

File: src/Lykke.Service.ClientSearch/Modules/ServiceModule.cs (hunk 4 above)
Comment (igromyko) [reply src/Lykke.Service.ClientSearch/Modules/ServiceModule.cs]: rewritten

File: src/Lykke.Service.ClientSearch/Startup.cs (hunk 3 above)
Comment (igromyko) [reply src/Lykke.Service.ClientSearch/Startup.cs]: moved in to new IndexerManager class

**Review by MikhailGerasimov — APPROVED** [review]

**Pull request merged**